pkg/datapath/loader/base.go

```diff
@@ -110,6 +110,43 @@ func writePreFilterHeader(preFilter *prefilter.PreFilter, dir string) error {
 	return fw.Flush()
 }
 
+type setting struct {
+	name      string
+	val       string
+	ignoreErr bool
+}
+
+func addENIRules(sysSettings *[]setting) error {
```

Comment:
Nit: While passing the slice "by reference" gets the job done, instead the pattern in Golang is to return the updated slice. Slices under-the-hood are just a pointer to an array, so there aren't copies of the data being made when passed by value. Only the slice header is copied, hence why we return the slice because it updates the header.

> To understand how slices work, it helps to understand how they are implemented. There is a little data structure, the slice header, that is the item associated with the slice variable, and that header describes a section of a separately allocated array. When we pass slice values around, the header gets copied but the array it points to is always shared.

Source: https://blog.golang.org/slices#Conclusion

Reply:
Thanks for the review, and for the feedback + link! I did have a doubt at some point, I guess I went C-style and just used a pointer... But that's fixed now. Interesting read!